### PR TITLE
New blog post for building a Buildah container image for Kubernetes

### PR DESCRIPTION
**Pull request by ipbabble**
Signed-off-by: William Henry @ipbabble <whenry@redhat.com>

### CODE REVIEW

File: source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+# buildah config --entrypoint /usr/bin/buildah $containerid
+```
+
+It’s also good practice, and pretty important, to set some other metadata like the image name and who created it etc.
+
+```
+# buildah config --author "wgh @ redhat.com" --created-by "ipbabble" --label name=buildah $containerid
+```
+
+Finally I need to commit the container to an image. It should be called `buildah` because it is good practice to name a command line tool container the name of the command. That way, if its ENTRYPOINT is set correctly, you can run it similar to the command it is containerizing. The following command ran a little over a minute and currently has not status output. So be patient.
```
Comment (TomSweeneyRedHat) [inline source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: "not status output" to "no status output"

File: source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+* You probably need network access if you are going to run something like `buildah bud`. You don't need this if you are just running `buildah images/containers` etc. But many use cases will be using `bud` and so I've just used the hosts network and now the bridge because it's a short running build process. So for the example I'll use the `--network host` parameter for `podman run`.
+* You'll need to bind mount an area of the file system, so that `buildah bud` can build up and then commit the image to the host file system. This can be `/var/lib/containers` if you like, so as it’s immediately shared shared on the hose, or it can be some build sandbox area etc. You also need to bind mount the Dockerfile path so that Buildah can see the Dockerfile. So for the example I’ll use the bind mount parameter `-v` or `--volume` parameter for `podman run`.
```
Comment (TomSweeneyRedHat) [inline source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: "on the hose" to "on the host"

File: source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+# podman run --privileged --network host -v /var/lib/containers:/var/lib/containers:rw  -v /home/whenry/dfs:/tmp:Z buildah bud -f /tmp/Dockerfile -t nginx 
+```
+
+Currently, due to OverlayFS requiring root privileges, this container must run in privileged mode. This requirement should be deprecated when Fedora’s OverlayFS does not require a privileged user.
+
+## Current issues with non privileged mode
+
+There is a requirement in large clusters for buildah not to run in privileged mode. There are ways to mitigate this but there will be cases where people demand it.
+
+I had prepared an example of how to do this but I will keep it for a later blog.  Here are some of the current constraints I encountered trying to solve the non-privileged problem
```
Comment (TomSweeneyRedHat) [inline source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: "non-privileged problem".  Not sure if you want an ':' or an '.' here, but I think you want one of the other at the end.

File: source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+* If you want to share the builds immediately through the hosts `/var/lib/containers` then you need to run Buildah as privileged. You could build somewhere else and then move the images to `/var/lib/containers` - see below. 
+* If you don't care to share on the host immediately then you can use unprivileged (don’t use --privileged) but you need to use a different directory than /var/lib/containers. However currently OverlayFS requires privileged mode on Fedora/RHEL. This should be fixed soon. You could use `--storage-driver=vfs`.
+* If you decide to build inside the container you can do that unpriv but you hit the overlay on overlay issue and then need to run with storage-driver vfs. This worked at one point but I’ve seen a regress.
```
Comment (TomSweeneyRedHat) [inline source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: Perhaps: "unpriv but you hit the overlay on overlay issue and then need to run with storage-driver vfs" to "unpriv but you will likely hit the overlay on overlay issue and will then need to run with storage-driver vfs"

File: source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md (hunk 1 above)
Comment (ipbabble) [reply source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: +1

File: source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md (hunk 2 above)
Comment (ipbabble) [reply source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: could have been worse. +1

File: source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md (hunk 3 above)
Comment (ipbabble) [reply source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: +1 '.'

File: source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md (hunk 4 above)
Comment (ipbabble) [reply source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: +1 changed to 
unprivileged, but you hit the overlay on overlay issue and will then need to run with `--storage-driver vfs`

Comment (TomSweeneyRedHat) [reply source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md]: :+1:

**Pull request merged**